# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/JitFrames.h
# Commit: 1c35c1a7b373
# Full Hash: 1c35c1a7b3731ecbc70be235430a6f06376494d1
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2022-06-09 22:00:48
# Regressor Bug: 1770366
# File Overlap Count: 2
# Description:
#   Bug 1770366 part 5 - Add frame pointer to IonJS/Warp frames. r=iain
#   
#   Change OSR to use the standard frame epilogue on the Baseline side and frame prologue
#   on the CodeGenerator side.
#   
# ==============================================================================

diff -r e8f787c59185 -r 1c35c1a7b373 js/src/jit/JitFrames.h
--- a/js/src/jit/JitFrames.h	Thu Jun 09 13:49:16 2022 +0000
+++ b/js/src/jit/JitFrames.h	Thu Jun 09 13:49:17 2022 +0000
@@ -276,6 +276,17 @@
   }
   uintptr_t numActualArgs() const { return numActualArgs_; }
 
+  void* callerFramePtr() const {
+    // The caller's frame pointer is pushed after the JitFrameLayout.
+    auto* p = reinterpret_cast<const uintptr_t*>(this) - 1;
+    return reinterpret_cast<void*>(*p);
+  }
+
+  // For IonJS frames: the distance from the JitFrameLayout to the first local
+  // slot. The caller's frame pointer is stored in this space. 32-bit platforms
+  // have 4 bytes of padding to ensure doubles are properly aligned.
+  static constexpr size_t IonFirstSlotOffset = 8;
+
   // Computes a reference to a stack or argument slot, where a slot is a
   // distance from the base frame pointer, as would be used for LStackSlot
   // or LArgument.